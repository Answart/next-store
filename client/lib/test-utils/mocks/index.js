export {
  mockUser, mockUsers,
  mockImage, mockImageVariables,
  mockProduct, mockProducts, mockShopProductsVariables,
  mockVariant,
  mockCartItem,
  mockOrderItem, mockOrder,
} from './typeDefs';
export {
  localUserQueryEmptyCartMock, localUserQueryNoUserMock, localUserQueryCartItemMock,
  userQueryEmptyCartMock, userQueryNoUserMock, userQueryCartItemMock,
  signupMutationMock,
  signinMutationMock,
  signoutMutationMock,
  requestPasswordResetMutationMock, requestPasswordResetMutationErrorMock,
  resetPasswordMutationMock, resetPasswordMutationErrorMock,
  updatePermissionsMutationMock,
  createImageMutationMock, createImageMutationErrorMock,
  productQueryMock, productQueryErrorMock, productQueryNoProductMock, productQueryNoVariantMock,
  shopProductsQueryFilterMock,
  createProductMutationMock, createProductMutationErrorMock,
  updateProductMutationMock, updateProductMutationErrorMock,
  deleteProductMutationMock,
  createProductVariantMutationMock, createProductVariantMutationErrorMock,
  updateProductVariantMutationMock, updateProductVariantMutationErrorMock,
  deleteProductVariantMutationMock,
  addToCartMutationMock,
  updateCartItemMutationMock,
  removeFromCartMutationMock,
} from './resolvers';
