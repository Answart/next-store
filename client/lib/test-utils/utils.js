import casual from 'casual';


casual.seed(777);

const fakeUser = () => ({
  __typename: 'User',
  id: 'us3r1d',
  name: casual.name,
  email: casual.email,
  password: 'unique-password',
  permissions: ['ADMIN'],
  cart: [],
});

const fakeImage = () => ({
  __typename: 'Image',
  id: '1m4g31d',
  cloudinary_id: "111111",
  name: "peggswatch1",
  width: 22,
  height: 22,
  transformation: "",
  image_url: "peggswatch1.jpg",
  large_image_url: "peggswatch2.jpg",
});

const fakeProduct = () => ({
  __typename: 'Product',
  id: 'pr0duct1d',
  department: "accessories",
  title: "Peggs Gold Edition Analog Watch Peggs Gold Edition Analog ",
  description: "Limited Edition watch from the 2018 Fall fashion line.",
  category: "jewelry",
  brand: "Peggs",
  online: false,
  user: fakeUser(),
  variants: [],
  image: fakeImage(),
});

const fakeVariant = () => ({
  __typename: 'Variant',
  id: 'v4r13nt1d',
  availability: "4 in Stock!",
  color: "white",
  price: 35,
  quantity: 4,
  sale: true,
  salePrice: 30,
  size: "S",
  image: fakeImage(),
  product: fakeProduct()
});

const fakeCartItem = overrides => ({
  __typename: 'CartItem',
  id: 'c4rt1t3m1d',
  quantity: 3,
  user: fakeUser(),
  variant: fakeVariant(),
  ...overrides,
});

const fakeOrder = () => ({
  __typename: 'Order',
  id: '0rd3r1d',
  charge: 'ch_123',
  total: 4000,
  status: 'processing',
  quantity: 3,
  subtotal: 20,
  shipping: 1.5,
  tax: 1.02,
  total: 20,
  sales_tax_rate: 1.02,
  shipping_rate: 1.02,
  payment: 'some payment',
  items: [],
  buyer: fakeUser(),
  createdAt: '2018-04 - 06T19: 24: 16.000Z',
  updatedAt: '2018-04 - 06T19: 24: 16.000Z',
});

const fakeOrderItem = () => ({
  __typename: 'OrderItem',
  id: '0rd3r1t3m1d',
  title: 'order item title',
  price: 12,
  quantity: 2,
  color: 'green',
  size: 'L',
  image_url: 'some_img_url',
  seller: fakeUser(),
  variant: fakeVariant(),
  order: fakeOrder(),
  createdAt: '2018-04 - 06T19: 24: 16.000Z',
  updatedAt: '2018-04 - 06T19: 24: 16.000Z',
});


export {
  fakeUser,
  fakeImage,
  fakeProduct,
  fakeVariant,
  fakeCartItem,
  fakeOrder,
  fakeOrderItem,
};
