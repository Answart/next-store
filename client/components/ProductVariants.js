import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { StyledProductVariants } from './styles/ProductStyles';
import PriceTag from './PriceTag';
import { getUniqKeyVals, getFltrdObjs, formatMoney, capWord } from '../lib/utils';


class ProductVariants extends Component {
  static propTypes = {
    variants: PropTypes.arrayOf(PropTypes.shape({
      size: PropTypes.string,
      color: PropTypes.string,
      availability: PropTypes.string,
      price: PropTypes.number,
      sale: PropTypes.bool.isRequired,
      salePrice: PropTypes.number
    })).isRequired,
    online: PropTypes.bool.isRequired,
    demoView: PropTypes.bool,
    VariantActionComponent: PropTypes.func,
    variantAction: PropTypes.func,
    variantActionLabel: PropTypes.string
  };
  constructor(props) {
    super(props);
    let newState = this.getStartState();
    const sizes = getUniqKeyVals(this.props.variants, 'size');
    newState.sizes = sizes;

    this.state = newState;
  }
  getStartState = (filterQuery = {}) => {
    let variants = this.props.variants || [];
    let currentVariants = [];
    let variant;

    let updating = Object.keys(filterQuery).length ? true : false;
    if (variants.length) variant = variants[0];

    if (variant && !updating) {
      if (variant.size) filterQuery.size = variant.size;
    }

    currentVariants = getFltrdObjs(variants, filterQuery);

    const colors = getUniqKeyVals(currentVariants, 'color');

    if (variant && !updating) {
      if (variant.color) filterQuery.color = variant.color;
    }

    if (filterQuery.color) currentVariants = getFltrdObjs(variants, filterQuery);

    variant = currentVariants.length ? currentVariants[0] : null;

    return {
      filterQuery,
      currentVariants,
      variant,
      colors
    };
  }
  updateFilter = e => {
    if (!!e && e.preventDefault) e.preventDefault();
    const { name, value } = e.currentTarget;
    let { filterQuery } = this.state;
    delete filterQuery[name];
    if (name && value.length) filterQuery[name] = value;
    if (name === 'size') delete filterQuery['color'];

    const state = this.getStartState(filterQuery);
    this.setState(state);
  }
  render() {
    const { sizes, colors, variant } = this.state;
    const {
      VariantActionComponent,
      variantAction, variantActionLabel,
      online, demoView
    } = this.props;
    const live = online && !demoView
    const accessible = (demoView || live);
    let availability = accessible && variant
      ? variant.availability
      : "Unavailable";
    return (
      <StyledProductVariants>
        {variant && (
          <div className="prdct-padding">
            <PriceTag price={variant.price} sale={variant.sale} salePrice={variant.salePrice} />
          </div>
        )}

        {sizes && !!sizes.length && !!sizes[0] && (
          <div id='prod-var-sizes' className="prdct-padding">
            <div className="prdct-padding filter-sizes">
              {sizes.map((size, i) => (
                <button key={i} id={`prod-var-size-${size}`} className={size == variant.size ? 'undrln-btn sample-selected' : 'undrln-btn sample-hover'}
                  name="size"
                  value={size}
                  title={`Select size: ${capWord(size)}`}
                  onClick={this.updateFilter}
                >
                  {capWord(size)}
                </button>
              ))}
            </div>
          </div>
        )}

        {colors && !!colors.length && !!colors[0] && (
          <div id='prod-var-colors' className="prdct-padding">
            <div id='prod-var-colors-title'>
              {!variant.color ? 'Colors:' : capWord(variant.color)}
            </div>

            <div className="prdct-padding filter-colors">
              {colors.map((color, i) => (
                <button key={i} id={`prod-var-color-${color}`} className={color == variant.color ? 'sample-selected' : 'sample-hover'}
                  name="color"
                  value={color}
                  title={`Select color: ${capWord(color)}`}
                  onClick={this.updateFilter}
                >
                  <div className={`color-sphere-sample ${color}-color-sample`}></div>
                </button>
              ))}
            </div>
          </div>
        )}

        <div id='prod-var-avail' className="prdct-padding">
          <i>
            {availability}
          </i>
        </div>

        {!!variant && accessible && (
          <div id='prod-var-action' className="prdct-padding">
            {(!!VariantActionComponent) && (
              <VariantActionComponent
                disabled={!variant.id}
                variant={variant}
              />
            )}
            {(!!variantAction && !!variantActionLabel) && (
              <button id='prod-var-action-btn' className="big-btn"
                disabled={!variant || !variant.id}
                onClick={(e) => variantAction(e, variant)}
              >
                {variantActionLabel}
              </button>
            )}
          </div>
        )}
      </StyledProductVariants>
    );
  }
}


export default ProductVariants;
