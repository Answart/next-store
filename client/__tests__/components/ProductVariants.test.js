import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import wait from 'waait';
import { MockedProvider } from 'react-apollo/test-utils';
import ProductVariants from '../../components/ProductVariants';
import {
  mockProduct, mockVariant
} from '../../lib/test-utils/mocks';

const ComponentToMock = (props) => (
  <div>A mock with '{Object.keys(props).length}' props passed!</div>
);
const variants = [{
  ...mockVariant
}, {
  ...mockVariant,
  id: '4352624623',
  color: 'green',
  size: 'S',
}, {
  ...mockVariant,
  id: '84202615159781',
  color: 'white',
  size: 'L',
}, {
  ...mockVariant,
  id: '9182737465',
  color: 'green',
  size: 'L',
}];


describe('<ProductVariants />', () => {
  describe('component w/variantAction and Label', () => {
    let wrapper;
    const variantAction = jest.fn();
    beforeAll(() => {
      wrapper = mount(
        <MockedProvider mocks={[]}>
          <ProductVariants
            variants={variants}
            online={true}
            demoView={true}
            VariantActionComponent={null}
            variantAction={variantAction}
            variantActionLabel='Select'
          />
        </MockedProvider>
      );
    });
    afterAll(() => {
      wrapper.unmount();
      jest.clearAllMocks();
    });

    it('matches the snap shot', async () => {
      await wait();
      wrapper.update();
      expect(toJSON(wrapper.find('ProductVariants'))).toMatchSnapshot();
    });

    it('renders properly', async () => {
      expect(wrapper.find('PriceTag').length).toBe(1);
      expect(wrapper.find('PriceTag').text()).toBe('$35$30');
      expect(wrapper.find('#prod-var-sizes').length).toBe(1);
      expect(wrapper.find('#prod-var-sizes').text()).toBe('SL');
      expect(wrapper.find('#prod-var-sizes').find('button').length).toBe(2);
      expect(wrapper.find('#prod-var-colors').length).toBe(1);
      expect(wrapper.find('#prod-var-colors').text()).toBe('White');
      expect(wrapper.find('#prod-var-colors').find('button').length).toBe(2);
      expect(wrapper.find('#prod-var-avail').length).toBe(1);
      expect(wrapper.find('#prod-var-avail').text()).toBe('4 in Stock!');
      expect(wrapper.find('#prod-var-action').length).toBe(1);
      expect(wrapper.find('VariantActionComponent').length).toBe(0);
      expect(wrapper.find('button#prod-var-action-btn').length).toBe(1);
      expect(wrapper.find('button#prod-var-action-btn').text()).toBe('Select');
    });

    it('clicking a color option changes its class', async () => {
      const buttons = wrapper.find('#prod-var-colors').find('button');
      const whiteBtnProps = buttons.at(0).props();
      const greenBtnProps = buttons.at(1).props();
      expect(whiteBtnProps.name).toBe('color');
      expect(whiteBtnProps.value).toBe('white');
      expect(whiteBtnProps.title).toBe('Select color: White');
      expect(whiteBtnProps.className).toBe('sample-selected');
      expect(greenBtnProps.name).toBe('color');
      expect(greenBtnProps.value).toBe('green');
      expect(greenBtnProps.title).toBe('Select color: Green');
      expect(greenBtnProps.className).toBe('sample-hover');
      buttons.at(1).simulate('click');
      await wait(100);
      wrapper.update();
      const buttons2 = wrapper.find('#prod-var-colors').find('button');
      expect(buttons2.at(0).props().className).toBe('sample-selected');
    });

    it('clicking a size option changes its class', async () => {
      const buttons = wrapper.find('#prod-var-sizes').find('button');
      const sBtn = buttons.at(0);
      const lBtn = buttons.at(1);
      expect(sBtn.text()).toBe('S');
      expect(sBtn.props().className).toBe('undrln-btn sample-selected');
      expect(lBtn.text()).toBe('L');
      expect(lBtn.props().className).toBe('undrln-btn sample-hover');
      buttons.at(1).simulate('click');
      await wait(100);
      wrapper.update();
      const buttons2 = wrapper.find('#prod-var-sizes').find('button');
      expect(buttons2.at(0).props().className).toBe('undrln-btn sample-hover');
      expect(buttons2.at(1).props().className).toBe('undrln-btn sample-selected');
    });

    it('action button clicks properly', async () => {
      const button = wrapper.find('button#prod-var-action-btn');
      expect(button.props().disabled).toBe(false);
      expect(button.text()).toBe('Select');
      expect(variantAction).not.toHaveBeenCalled();
      button.simulate('click');
      await wait(50);
      wrapper.update();
      expect(variantAction).toHaveBeenCalled();
    });
  });

  describe('component w/VariantActionComponent', () => {
    let wrapper2;
    const variantActionComponent = jest.fn();
    beforeAll(() => {
      wrapper2 = mount(
        <MockedProvider mocks={[]}>
          <ProductVariants
            variants={variants}
            online={true}
            demoView={true}
            VariantActionComponent={ComponentToMock}
            variantAction={null}
            variantActionLabel=''
          />
        </MockedProvider>
      );
    });
    afterAll(() => {
      wrapper2.unmount();
      jest.clearAllMocks();
    });

    it('renders properly', async () => {
      await wait();
      wrapper2.update();
      expect(wrapper2.find('PriceTag').length).toBe(1);
      expect(wrapper2.find('PriceTag').text()).toBe('$35$30');
      expect(wrapper2.find('#prod-var-sizes').length).toBe(1);
      expect(wrapper2.find('#prod-var-sizes').find('button').length).toBe(2);
      expect(wrapper2.find('#prod-var-colors').length).toBe(1);
      expect(wrapper2.find('#prod-var-colors').find('button').length).toBe(2);
      expect(wrapper2.find('#prod-var-avail').length).toBe(1);
      expect(wrapper2.find('#prod-var-avail').text()).toBe('4 in Stock!');
      expect(wrapper2.find('#prod-var-action').length).toBe(1);
      expect(wrapper2.find('ComponentToMock').length).toBe(1);
      expect(wrapper2.find('ComponentToMock').props().disabled).toBe(false);
      expect(wrapper2.find('ComponentToMock').props().variant).toMatchObject(mockVariant);
      expect(wrapper2.find('button#prod-var-action-btn').length).toBe(0);
    });
  });
});
