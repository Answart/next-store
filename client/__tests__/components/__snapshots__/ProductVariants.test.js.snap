// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductVariants /> component w/variantAction and Label matches the snap shot 1`] = `
<ProductVariants
  VariantActionComponent={null}
  demoView={true}
  online={true}
  variantAction={[MockFunction]}
  variantActionLabel="Select"
  variants={
    Array [
      Object {
        "__typename": "Variant",
        "availability": "4 in Stock!",
        "color": "white",
        "id": "v4r13nt1d",
        "image": Object {
          "__typename": "Image",
          "cloudinary_id": "111111",
          "height": 22,
          "id": "1m4g31d",
          "image_url": "peggswatch1.jpg",
          "large_image_url": "peggswatch2.jpg",
          "name": "peggswatch1",
          "transformation": "",
          "width": 22,
        },
        "price": 35,
        "product": Object {
          "__typename": "Product",
          "brand": "Peggs",
          "category": "jewelry",
          "department": "accessories",
          "description": "Limited Edition watch from the 2018 Fall fashion line.",
          "id": "pr0duct1d",
          "image": Object {
            "__typename": "Image",
            "cloudinary_id": "111111",
            "height": 22,
            "id": "1m4g31d",
            "image_url": "peggswatch1.jpg",
            "large_image_url": "peggswatch2.jpg",
            "name": "peggswatch1",
            "transformation": "",
            "width": 22,
          },
          "online": false,
          "title": "Peggs Gold Edition Analog Watch Peggs Gold Edition Analog ",
          "user": Object {
            "__typename": "User",
            "cart": Array [],
            "email": "Princess.Ernser@Dibbert.net",
            "id": "us3r1d",
            "name": "Dr. Randy Gleason",
            "password": "unique-password",
            "permissions": Array [
              "ADMIN",
            ],
          },
          "variants": Array [],
        },
        "quantity": 4,
        "sale": true,
        "salePrice": 30,
        "size": "S",
      },
      Object {
        "__typename": "Variant",
        "availability": "4 in Stock!",
        "color": "green",
        "id": "4352624623",
        "image": Object {
          "__typename": "Image",
          "cloudinary_id": "111111",
          "height": 22,
          "id": "1m4g31d",
          "image_url": "peggswatch1.jpg",
          "large_image_url": "peggswatch2.jpg",
          "name": "peggswatch1",
          "transformation": "",
          "width": 22,
        },
        "price": 35,
        "product": Object {
          "__typename": "Product",
          "brand": "Peggs",
          "category": "jewelry",
          "department": "accessories",
          "description": "Limited Edition watch from the 2018 Fall fashion line.",
          "id": "pr0duct1d",
          "image": Object {
            "__typename": "Image",
            "cloudinary_id": "111111",
            "height": 22,
            "id": "1m4g31d",
            "image_url": "peggswatch1.jpg",
            "large_image_url": "peggswatch2.jpg",
            "name": "peggswatch1",
            "transformation": "",
            "width": 22,
          },
          "online": false,
          "title": "Peggs Gold Edition Analog Watch Peggs Gold Edition Analog ",
          "user": Object {
            "__typename": "User",
            "cart": Array [],
            "email": "Princess.Ernser@Dibbert.net",
            "id": "us3r1d",
            "name": "Dr. Randy Gleason",
            "password": "unique-password",
            "permissions": Array [
              "ADMIN",
            ],
          },
          "variants": Array [],
        },
        "quantity": 4,
        "sale": true,
        "salePrice": 30,
        "size": "S",
      },
      Object {
        "__typename": "Variant",
        "availability": "4 in Stock!",
        "color": "white",
        "id": "84202615159781",
        "image": Object {
          "__typename": "Image",
          "cloudinary_id": "111111",
          "height": 22,
          "id": "1m4g31d",
          "image_url": "peggswatch1.jpg",
          "large_image_url": "peggswatch2.jpg",
          "name": "peggswatch1",
          "transformation": "",
          "width": 22,
        },
        "price": 35,
        "product": Object {
          "__typename": "Product",
          "brand": "Peggs",
          "category": "jewelry",
          "department": "accessories",
          "description": "Limited Edition watch from the 2018 Fall fashion line.",
          "id": "pr0duct1d",
          "image": Object {
            "__typename": "Image",
            "cloudinary_id": "111111",
            "height": 22,
            "id": "1m4g31d",
            "image_url": "peggswatch1.jpg",
            "large_image_url": "peggswatch2.jpg",
            "name": "peggswatch1",
            "transformation": "",
            "width": 22,
          },
          "online": false,
          "title": "Peggs Gold Edition Analog Watch Peggs Gold Edition Analog ",
          "user": Object {
            "__typename": "User",
            "cart": Array [],
            "email": "Princess.Ernser@Dibbert.net",
            "id": "us3r1d",
            "name": "Dr. Randy Gleason",
            "password": "unique-password",
            "permissions": Array [
              "ADMIN",
            ],
          },
          "variants": Array [],
        },
        "quantity": 4,
        "sale": true,
        "salePrice": 30,
        "size": "L",
      },
      Object {
        "__typename": "Variant",
        "availability": "4 in Stock!",
        "color": "green",
        "id": "9182737465",
        "image": Object {
          "__typename": "Image",
          "cloudinary_id": "111111",
          "height": 22,
          "id": "1m4g31d",
          "image_url": "peggswatch1.jpg",
          "large_image_url": "peggswatch2.jpg",
          "name": "peggswatch1",
          "transformation": "",
          "width": 22,
        },
        "price": 35,
        "product": Object {
          "__typename": "Product",
          "brand": "Peggs",
          "category": "jewelry",
          "department": "accessories",
          "description": "Limited Edition watch from the 2018 Fall fashion line.",
          "id": "pr0duct1d",
          "image": Object {
            "__typename": "Image",
            "cloudinary_id": "111111",
            "height": 22,
            "id": "1m4g31d",
            "image_url": "peggswatch1.jpg",
            "large_image_url": "peggswatch2.jpg",
            "name": "peggswatch1",
            "transformation": "",
            "width": 22,
          },
          "online": false,
          "title": "Peggs Gold Edition Analog Watch Peggs Gold Edition Analog ",
          "user": Object {
            "__typename": "User",
            "cart": Array [],
            "email": "Princess.Ernser@Dibbert.net",
            "id": "us3r1d",
            "name": "Dr. Randy Gleason",
            "password": "unique-password",
            "permissions": Array [
              "ADMIN",
            ],
          },
          "variants": Array [],
        },
        "quantity": 4,
        "sale": true,
        "salePrice": 30,
        "size": "L",
      },
    ]
  }
>
  <ProductStyles__StyledProductVariants>
    <div
      className="ProductStyles__StyledProductVariants-sc-2231u4-1 ndRtn"
    >
      <div
        className="prdct-padding"
      >
        <PriceTag
          price={35}
          sale={true}
          salePrice={30}
        >
          <PriceTag__StyledPriceTag>
            <div
              className="PriceTag__StyledPriceTag-sc-119rsfa-0 ePOheW"
            >
              <span
                className="line-through"
              >
                $35
              </span>
              <span
                className="price-tag-padding price-tag-priceSale"
              >
                $30
              </span>
            </div>
          </PriceTag__StyledPriceTag>
        </PriceTag>
      </div>
      <div
        className="prdct-padding"
        id="prod-var-sizes"
      >
        <div
          className="prdct-padding filter-sizes"
        >
          <button
            className="undrln-btn sample-selected"
            id="prod-var-size-S"
            key="0"
            name="size"
            onClick={[Function]}
            title="Select size: S"
            value="S"
          >
            S
          </button>
          <button
            className="undrln-btn sample-hover"
            id="prod-var-size-L"
            key="1"
            name="size"
            onClick={[Function]}
            title="Select size: L"
            value="L"
          >
            L
          </button>
        </div>
      </div>
      <div
        className="prdct-padding"
        id="prod-var-colors"
      >
        <div
          id="prod-var-colors-title"
        >
          White
        </div>
        <div
          className="prdct-padding filter-colors"
        >
          <button
            className="sample-selected"
            id="prod-var-color-white"
            key="0"
            name="color"
            onClick={[Function]}
            title="Select color: White"
            value="white"
          >
            <div
              className="color-sphere-sample white-color-sample"
            />
          </button>
          <button
            className="sample-hover"
            id="prod-var-color-green"
            key="1"
            name="color"
            onClick={[Function]}
            title="Select color: Green"
            value="green"
          >
            <div
              className="color-sphere-sample green-color-sample"
            />
          </button>
        </div>
      </div>
      <div
        className="prdct-padding"
        id="prod-var-avail"
      >
        <i>
          4 in Stock!
        </i>
      </div>
      <div
        className="prdct-padding"
        id="prod-var-action"
      >
        <button
          className="big-btn"
          disabled={false}
          id="prod-var-action-btn"
          onClick={[Function]}
        >
          Select
        </button>
      </div>
    </div>
  </ProductStyles__StyledProductVariants>
</ProductVariants>
`;
